### PR TITLE
OSSL_STORE: Fix crash when tracing STORE

### PR DESCRIPTION
**Pull request by levitte**

### CODE REVIEW

File: crypto/store/store_lib.c [hunk 1]
```diff
@@ -111,9 +111,10 @@ OSSL_STORE_open_ex(const char *uri, OSSL_LIB_CTX *libctx, const char *propq,
      * elsewhere.
      */
     for (i = 0; loader_ctx == NULL && i < schemes_n; i++) {
-        OSSL_TRACE1(STORE, "Looking up scheme %s\n", schemes[i]);
+        scheme = schemes[i];
+        OSSL_TRACE1(STORE, "Looking up scheme %s\n", scheme);
```
Comment (slontis) [inline crypto/store/store_lib.c:115]: How was this causing a crash?

Comment (levitte) [reply crypto/store/store_lib.c:115]: Further down, old line 154, would crash because `i == schemes_n`, i.e. a classic fence post error.

Comment (slontis) [reply crypto/store/store_lib.c:115]: Ta - couldnt see the end of the for loop here...

**Review by slontis — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by levitte** [comment]
Merged

19c0b46b83335b36a9816abef4e82f74863a4e0a OSSL_STORE: Fix crash when tracing STORE

**Comment by levitte** [comment]
This has me wonder if we should have a test that tries the diverse trace flags...  or all.  Like running the whole test suite with the environment variable `OPENSSL_TRACE=ALL`...

**Pull request closed without merging**